transcom/mymove
# B 19926 loa lookup UI int

## [B-19926](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A948238)

## Summary

Introduces long line of accounting display and concatenation on the orders page for office users. Includes a warning for missing / invalid TAC.

The state of the LOA preview should change when:

- The TAC is updated and is equal to a length of 4
- The issue date changes
- The department changes

### How to test

1. Log into milmovelocal
2. Create a new move and HHG shipment
3. Write down the move locator
4. Log into officelocal as a service counselor
5. Navigate to the move -> edit orders
6. Fill out the orders like so:
  1. Issue date: 2014-01-01
  2. Department: ARMY
  3. TAC: E12A
7. Upon filling out the above, the LOA preview at the bottom should populate (Subject to change based on randomly generated seed data)
8. Save the form if you see the LOA preview populate
9. Switch to TIO/TOO
10. Navigate to the move
11. Open edit orders -> the LOA should populate successfully without you touching anything

## Screenshots

![image](https://github.com/transcom/mymove/assets/136814362/d59d2763-3119-4c87-a819-0fa024147437)
 
![image](https://github.com/transcom/mymove/assets/136814362/51bac083-6165-449b-95f3-2815675bfd72)

![image](https://github.com/transcom/mymove/assets/136814362/a9203c1c-1fb9-4126-b5ed-5ff357f467a2)
